File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int> temp;
        int count = 0;
     for(int i = 0; i < nums.size();i++)
     {
        if (nums[i] != 0)
        {
            temp.push_back(nums[i]);
            count++;
        }
      
     } 
     for(int i = count ; i< nums.size(); i++)
     {
        temp.push_back(0);
     }  
     for(int i = 0; i< nums.size(); i++)
     {
        nums[i] = temp[i];
     }
    }
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n=nums.length;
        if(n==1){
            return ;
        }
        int a=0;
        int b=0;
        for(int i=0;i<n;i++){
            if(nums[a]==0 && nums[i]!=0){
                nums[a]=nums[i];
                nums[i]=0;
                a++;
                
            }
            if(nums[a]!=0){
                a++;
            }
           

        }
        
        
    }
}